### ISSUE TITLE
Support Select entities and create seperate devices for each "LP" and more...

### ISSUE THREAD

**Issue by Bascht74** [a529987659852/openwbmqtt#17]
Hi, 

I tried the current version of your HA integration and is very nice and easy to install. Very nice!

I have some suggestions for the next versions... (sorted by importance, e.g. 4-7 are more like "nice to have")

1. If you have two or more "LP" it makes more sense to separate the entities for the two LP in different devices. So one device for each LP (`OpenWB {name} (LP1) `, `OpenWB {name} (LP2)` a.s.o `{name}` from `openWB/lp/{n}/strChargePointName`). The unique entities (that exists only once on the openWB controller) could be bound to a "general" device named e.g. `OpenWB Controller` (e.g. sensor.openwb_leistungsabgabe_haus_pv). The `entities sensor.openwb_cp{n}_ladepunktsbezeichnung` could be deleted as they are not needed anymore...
2. Use the "Select" entity instead of the sensor entity for `openwb_lademodus`, so you could get rid of the service `change_global_charge_mode`. If you use number entities and select entities you could substitute every service that is needed right now and the entities are way more usable for users.
3. The entities with "Wh" sould be in "kWh" (e.g. `openwb_energie_erzeugt_haus`, this is more readable and uses the same unit as energy meters and the LP itself (e.g. `sensor.openwb_cp1_geladene_energie` and `sensor.openwb_cp1_ladezahler` uses kWh).
4. Instead of "`true`" and "`false`" you should use "`on`" and "`off`" for the binary sensors, e.G. `sensor.openwb_cp1_energiemengenbegrenzung_aktiv_modus_sofortladen` should be "`on`" or "`off`". If you use "`on`" and "`off`", HA translates the state to the language set in HA. It would be better as well to use `binary sensors` instead of `sensors` for those entities with `true` / `false`.
5. The IP-Address entity could be disabled or removed because you used the "visit" function.
6. You could read the ip address from `openWB/system/IpAddress` so it is not needed for setup and could be read at start of the integration.
7. The active LP could be read through a loop into `openWB/lp/{n}/` and check `boolChargePointConfigured`. If you combine 6+7 only  one parameter is needed for setup. The check for LP could be at integration start so that you can add new devices if a user installs e.g. a second LP.
8. show openWB/system/Version as firmware version like: 
![image](https://user-images.githubusercontent.com/66269310/148661804-0825a416-e918-4873-b77a-389ece81f1bb.png)

Thx again for the integration. I hope that constructive feedback is welcome...

Sebastian

**Comment by a529987659852** [comment]
Thank you very much for your feedback, Sebastian!
I'll take this into consideration when I start the next round of development. Due to the fact that I'm still waiting for my openWB to be shipped, my motivation to improve the code is not so high at the moment. So, this will take some time.

If you want to contribute in the meantime, feel free to make a PR which I'm happy to include.

Andreas

**Comment by northalpha** [comment]
Just as a side note: my openWB is finally here and HA integration is working together but i am waiting for my BEV and so i may can contribute by testing or so.
